brian2/core/functions.py

```diff
@@ -540,6 +541,27 @@ def eval(cls, args):
         return sympy.functions.elementary.exponential.log(args, 10)
 
 
+_iinfo = np.iinfo(int)
+
+def timestep(t, dt):
+    elapsed_steps = (t + 1e-3 * dt) / dt
+    if np.isscalar(elapsed_steps) or elapsed_steps.shape == ():
+        if np.isinf(elapsed_steps):
+            if elapsed_steps > 0:
+                return _iinfo.max
+            else:
+                return _iinfo.min
+        else:
+            return np.int_(elapsed_steps)
+    else:
+        are_inf = np.isinf(elapsed_steps)
+        int_steps = np.asarray(elapsed_steps, dtype=int)
+        if any(are_inf):
+            int_steps[are_inf & (elapsed_steps < 0)] = _iinfo.min
+            int_steps[are_inf & (elapsed_steps > 0)] = _iinfo.max
```

Comment:
Do we ever have t=+inf? Maybe we can leave this?

Reply:
Actually with the refractoriness we only ever have `+inf` since we pass `t - lastspike` and `lastspike` is `-inf` if no spike has occurred yet. So that would mean we could ignore the `-inf` case. But since the plan is to provide the `timestep` function to the user as well, I find it dangerous to ignore the other corner case. E.g. instead of doing `timestep(t - lastspke, dt)`, what if the user uses `timestep(t, dt) - timestep(lastspike, dt)`? Oh, actually that would lead to an overflow even with the correct implementation :-/ Maybe we should convert `inf` into something like `1e9` instead of the maximum integer value? Seems a bit arbitrary, though...

Reply:
I think in the C++ and Cython it's fine because it's just a branch that isn't usually taken, but it's a lot of work for numpy.

We could use +/- ``ourmaxint = min(intmax, -intmin)/2`` as the converted value so that arithmetic with it would work? We could also maybe set ``int_steps[are_inf] = sign(elapsed_steps[are_inf])*ourmaxint``? Might be even quicker to compute ``I,=are_inf.nonzero()`` first and then use that. Basically, at the moment we have to do N-element operations five times and we should only do that once since usually most ``are_inf`` will be false.